app/models/tag.rb

```diff
@@ -108,6 +108,10 @@ def tagged_documents
     @_tagged_documents ||= TaggedDocuments.new(self)
   end
 
+  def tagged_document_for_base_path(base_path)
+    tagged_documents.find { |document| document.base_path == base_path }
```

Comment:
if you trying to find the first match use `detect` instead of `find`

> Passes each entry in enum to block. Returns the first for which block is not false. If no object matches, calls ifnone and returns its result when it is specified, or returns nil otherwise.
> If no block is given, an enumerator is returned instead.
```
(1..10).detect   { |i| i % 5 == 0 and i % 7 == 0 }   #=> nil
(1..100).find    { |i| i % 5 == 0 and i % 7 == 0 }   #=> 35
```